pandas/util/testing.py

```diff
@@ -1403,7 +1424,11 @@ def assert_frame_equal(
             )
 
 
-def assert_equal(left, right, **kwargs):
+def assert_equal(
+    left: Union[DataFrame, AnyArrayLike],
```

Comment:
I think should use `FrameOrSeries` instead of `DataFrame`

Reply:
We want to cover the following types `Index, Series, DataFrame, ExtensionArray, or np.ndarray`. AnyArrayLike covers Index, Series, ExtensionArray and np.ndarray, so the only one missing is DataFrame. FrameOrSeries doesn't cover Index, ExtensionArray nor np.ndarray.

Reply:
Ah OK makes sense